plugins/outputs/http/http.go

```diff
@@ -168,6 +168,15 @@ func (h *HTTP) writeMetric(reqBody []byte) error {
 		req.SetBasicAuth(h.Username, h.Password)
 	}
 
+	// google api auth
+	if h.HTTPClientConfig.OAuth2Config.CredentialsFile != "" {
+		token, err := h.HTTPClientConfig.OAuth2Config.GetAccessToken(context.Background(), h.URL)
+		if err != nil {
+			return err
+		}
+		token.SetAuthHeader(req)
+	}
```

Comment:
HTTPClientConfig is also used by plugins/inputs/http and plugins/inputs/openstack. Would it make sense to move this section to a function and call it in those two plugins too?

Reply:
@zachmares what do you think about adding it to these other plugins? I don't really like the idea of adding an undocumented variable to other plugins if it won't ever be useful.

Reply:
@reimda Apologies, I missed this comment yesterday. I'm not entirely sure at the moment. I'd need to defer to the experts for this one.

Reply:
Hi @reimda @srebhan, 
This PR was really just for getting through the Google auth using the HTTP output plugin. 
I can see that it could make sense to make this available to other plugins, but being that this doesn't fall within the scope of the description or the issue it's trying to solve for I would say it should be a seperate PR.
Does that make sense?

Reply:
I'm not sure oauth makes sense in the context of a Telegraf input plugin. This contribution sets auth headers on the payloads written from the output plugin.

I'm happy to discuss and explore this further, but I agree with @crflanigan that those discussions are better suited for a future and separate PR. A use case would be helpful too. 

I believe this contribution is self-contained and ready to go.

Reply:
I understand that the scope of this pr is really only the http output, but adding the setting to the existing OAuth2Config struct has the side effect of also adding it to the http and openstack inputs. That makes the PR not ready to go.

Having CredentialsFile in inputs/http and inputs/openstack might not seem important, but are plans to add an api to query and change settings: #7519. That will expose the setting to users.

We could ignore the problem now, but if it turns out we need to remove it from OAuth2Config in the future because it doesn't apply to the inputs, it will not be possible to leave it in for the outputs. We want to plan for compatibility for the http output plugin and not add a setting we're likely to remove. It's better to find the right place for the setting now than rush this PR and get it wrong.

If it doesn't makes sense to add CredentialsFile to the http input and the openstack input, it should not be in OAuth2Config. If it is only going to be useful in the http output plugin, it should be in plugins/outputs/http/http.go's HTTP struct, not a shared struct like OAuth2Config or HTTPClientConfig.

Reply:
I can imagine there are APIs you want to pull with the http input that require Oauth2. I do agree not to implement it in input plugin for now as there is no need for it (yet?)

Reply:
I moved the Google API Auth settings and code to outputs/http.go as requested.

@reimda @srebhan Please take a look when you get the chance.

Reply:
After a short discussion on Slack, @zachmares agrees that this might also be helpful to at least `inputs.http`, so I suggest to keep the code in `common` and add the 
```golang
	// google api auth
	if h.HTTPClientConfig.OAuth2Config.CredentialsFile != "" {
		token, err := h.HTTPClientConfig.OAuth2Config.GetAccessToken(context.Background(), h.URL)
		if err != nil {
			return err
		}
		token.SetAuthHeader(req)
	}
```
part to both `http` input and output plugin at least.

Reply:
Sorry, seems like I got it wrong. @zachmares, can you please state if this auth method is useful for anything beyond `outputs.http` or not!? If it is *only* useful for `outputs.http` it should move there, otherwise, please keep it in common and add the above code to any plugin where it makes sense...

Sorry for the confusion I caused!

Reply:
No worries, Sven! As far as @crflanigan and I know, this contribution only makes sense in the outputs/http context. To recap, our use case is using Telegraf to gather system metrics and flushing them to an endpoint hosted in GCP where only authenticated invocations of the service are allowed.

If a use case arises wherein the an inputs plugin wishes to invoke a service in GCP, we'd be happy to move the code back to common.

Reply:
Although there are other output plugins that could benefit from this, such as `/plugins/outputs/bigquery` and `plugins/outputs/cloud_pubsub`...